drivers/adc/gecko_adc.c

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2022 T-Mobile USA, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_adc_gecko
+
+#include <string.h>
+#include <zephyr/drivers/adc/gecko_adc/gecko_adc.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/devicetree.h>
+
+#include "em_cmu.h"
+#include "em_gpio.h"
+
+K_SEM_DEFINE(adc_sem, 0, 1);
+
+static int get_reference_voltage(int reference)
+{
+	int ref = -1;
+
+	switch (reference) {
+	case ADC_REF_INTERNAL:
+		ref = adcRef2V5;
+		break;
+	case ADC_REF_VDD_1:
+		ref = adcRefVDD;
+		break;
+	case ADC_REF_VDD_1_2:
+		ref = adcRef2V5;
+		break;
+	case ADC_REF_VDD_1_4:
+		ref = adcRef1V25;
+		break;
+	default:
+		break;
```

Comment:
Setting the default (ref) return value (of -1) in the default switch case would be more evident than assigning it in the variable's declaration (above)